http-native/src/main/java/org/ballerinalang/net/http/HttpCallableUnitCallback.java

```diff
@@ -47,6 +51,13 @@
     public void notifySuccess(Object result) {
         if (result == null) { // handles nil return and end of resource exec
             requestMessage.waitAndReleaseAllEntities();
+            if (ObserveUtils.isObservabilityEnabled()) {
+                ObserverContext observerContext
+                        = (ObserverContext) requestMessage.getProperty(OBSERVABILITY_CONTEXT_PROPERTY);
+                if (observerContext != null) {
+                    ObserveUtils.stopObservationWithContext(observerContext);
```

Comment:
Don't we need to check whether the observation should be manually closed (`if (observerContext.isManuallyClosed())`) here before closing it ? (Same in the `sendFailureResponse` method as well)

Reply:
`observerContext` being present in the request message means it should be manually closed. Should I explicitly check it?

Reply:
Ahh yes. It should be enough :+1: